### DISCUSSION TITLE
Modules

### DISCUSSION THREAD

**Discussion by bitnom** [valkey-io/valkey#108] [Q&A]
What's the plan regarding the existing modules? Do they come along for the ride?

**Comment by madolson** [comment]
The existing modules available in Redis stack were always source available (Json, Search, Bloom, timeseries, etc). If there is some specific feature you are interested in, please open issue and we'll look into implementing it in a way that everyone can use it.

**Reply by natoscott** [reply]
There are also the forked versions of the re-licensed modules used by Debian and Fedora from the last round of 'source available' repo conversions - if we can avoid breaking these during this latest transition, it'd be greatly appreciated.

    https://github.com/goodform/

If it cannot be avoided, we can certainly fix these modules to move forward with valkey.

**Reply by madolson** [reply]
That's a good point, I had forgotten that the modules were at one point open source. I know the JSON module at the time of the fork was not good (had significant memory and performance issues) and would rather start from scratch. RedisGraph has one star at this point, which leads me to believe it was never really widely adopted. That leaves bloom and search as candidates.

**Reply by natoscott** [reply]
We are using the Search and JSON modules for some peripheral functionality on a project here, small number of keys and small values - both have been perfectly fine for this use-case.  Search vendors Bloom, so we indirectly use that too.

**Reply by PingXie** [reply]
I think it would make sense to start curating a module repo under the valkey-io project.

**Reply by zuiderkwast** [reply]
To start with, let's at least link to these on our README and/or docs!

**Reply by madolson** [reply]
I would love to see us have links for recommended modules as well as something similar to redis stack with our curated list of modules. (Maybe overlapping with ping's suggestion) 

Open source only? I.e. No rsal modules.

**Reply by natoscott** [reply]
Once the current API changes settle down and a release is available, I'd be happy to start moving these open source modules over to Valkey-specific module APIs and shift the git repos over here under the valkey.io umbrella - if there's consensus this is the way to go?

**Reply by rk94655** [reply]
When is the plan to move JSON module into valkey @natoscott

**Reply by natoscott** [reply]
You're referring to this module?  https://github.com/goodform/valkey-json 

I don't know.  :)  I plan some additional work to get the test harness running in an automated fashion for all the Valkey modules there, but it could transition at any time now I guess.  Ideally I think we'd switch and remove all git repos under the 'goodform' organisation completely, and just have these as valkey modules going forward.

**Comment by zuiderkwast** [comment]
We'll keep a compatible module API so you can run proprietary modules if you want. That said, I'd like to see the open ones if people are willing to contribute.

**Reply by natoscott** [reply]
I've migrated the first couple of modules (forked from before license change) over to Valkey APIs here:

https://github.com/natoscott/valkey-bloom
https://github.com/natoscott/valkey-json

These (as well as Search, and others) use two small helpers - test and SDK modules, which are here:

https://github.com/natoscott/valkey-module-test
https://github.com/natoscott/valkey-module-sdk

If this is the way you'd like to go, let me know and I can do the other modules as well.

**Reply by styk-tv** [reply]
such a shame on the  AGPL-3.0 license. unusable :(

**Reply by rabinnh** [reply]
Lack of a JSON module is the only thing keeping me from trying Valkey.  Changing all our code to work around not having that module is a big task with a ton of testing.

Curious: I don't see a response to the entry about the AGPL license.  Was that license applicable to the Redis module that was forked, or was that added to the valkey-json license?

**Reply by zuiderkwast** [reply]
There's an AGPL module forked from a an old version of the redis module when it was AGLP. That's not officially part of the valkey project though.

The official valkey module is going to be another implementation with another license, probably BSD like valkey itself. It's work in progress. Here is some text about it: https://github.com/valkey-io/valkey-rfc/blob/main/ValkeyJSON.md

**Comment by artikell** [comment]
I tried to organize the existing list of modules to help everyone find better ones. I feel that using the valkey name may not be easy to locate, so I used redis module to name it: https://github.com/goodform/awesome-redis-module

**Comment by mietzen** [comment]
Is there a valkey docker image that bundles the available modules like: redis/redis-stack-server ?

**Reply by zuiderkwast** [reply]
We will have a release bundle like that in the future.

**Reply by hassanbsalimi** [reply]
Any update on the availability of release bundle?
